tidy3d/components/simulation.py

```diff
@@ -5318,3 +5664,86 @@ def from_scene(cls, scene: Scene, **kwargs) -> Simulation:
         )
 
     _boundaries_for_zero_dims = validate_boundaries_for_zero_dims()
+
+    def _make_pec_frame(self, obj) -> Structure:
+        """Make a pec frame around a mode source."""
+        span_inds = np.array(self.grid.discretize_inds(obj))
+
+        coords = self.grid.boundaries.to_list
+        direction = obj.direction
+        if isinstance(obj, ModeSource):
```

Comment:
Right now, `obj` can be either a mod source, or port absorber?

Reply:
yes, added a type annotation for that